libs/features/richresults.js

```diff
@@ -1,26 +1,61 @@
-// createTag and getMetadata are passed in to avoid circular dependencies
-export function addRichResults(type, { createTag, getMetadata }) {
-  const addRichResultsForNewsArticle = () => {
-    const newsArticle = {
-      '@context': 'https://schema.org',
-      '@type': 'NewsArticle',
-      headLine: getMetadata('og:title'),
-      image: getMetadata('og:image'),
-      datePublished: getMetadata('published'),
-      dateModified: getMetadata('modified'),
-      author: {
-        '@type': 'Person',
-        name: getMetadata('authorname'),
-        url: getMetadata('authorurl'),
+function getRichResultsForNewsArticle(getMetadata) {
+  return {
+    '@context': 'https://schema.org',
+    '@type': 'NewsArticle',
+    headLine: getMetadata('og:title'),
+    image: getMetadata('og:image'),
+    datePublished: getMetadata('published'),
+    dateModified: getMetadata('modified'),
+    author: {
+      '@type': 'Person',
+      name: getMetadata('authorname'),
+      url: getMetadata('authorurl'),
+    },
+  };
+}
+
+function getRichResultsForSiteSearchBox(getMetadata) {
+  // See specifications at https://developers.google.com/search/docs/appearance/structured-data/sitelinks-searchbox
+  const SEARCH_TERM_STRING = 'search_term_string';
+  const urlTemplate = `${getMetadata('search-url')}?${getMetadata('search-parameter-name')}={${SEARCH_TERM_STRING}}`;
+  return {
+    '@context': 'https://schema.org',
+    '@type': 'WebSite',
+    url: getMetadata('url'),
+    potentialAction: {
+      '@type': 'SearchAction',
+      target: {
+        '@type': 'EntryPoint',
+        urlTemplate: urlTemplate,
       },
-    };
-    const script = createTag('script', { type: 'application/ld+json' }, JSON.stringify(newsArticle));
-    document.head.append(script);
-  }
-  
+      'query-input': `required name=${SEARCH_TERM_STRING}`,
+    },
+  };
+}
+
+function getRichResults(type, getMetadata) {
   switch (type) {
-    // add support for new types here, e.g. Product
     case 'NewsArticle':
-      addRichResultsForNewsArticle();
+      return getRichResultsForNewsArticle(getMetadata);
+    case 'SiteSearchBox':
+      return getRichResultsForSiteSearchBox(getMetadata);
+    default:
+      // eslint-disable-next-line no-console
+      console.error(`Type ${type} is not supported`);
+      return null;
   }
 }
+
+function addToDom(richResults, createTag) {
+  if (!richResults) {
+    return;
+  }
+  const script = createTag('script', { type: 'application/ld+json' }, JSON.stringify(richResults));
+  document.head.append(script);
+}
+
+// createTag and getMetadata are passed in to avoid circular dependencies
+export function addRichResults(type, { createTag, getMetadata }) {
```

Comment:
nit, to make the linter happy you'd have to add a default here: 
`export default function addRichResults(type, { createTag, getMetadata }) {`

Reply:
Addressed with https://github.com/adobecom/milo/pull/389/commits/4202377535cd937ec2c29bf8719c23dc4abda6e1